src/hooks/useCombobox/index.js

```diff
@@ -170,309 +174,339 @@ function useCombobox(userProps = {}) {
     },
   )
   /* Event handler functions */
-  const inputKeyDownHandlers = {
```

Comment:
Most of everything below here is formatting changes due to memoization. It will be minorly easier to review this [with whitespace changes ignored](https://github.com/downshift-js/downshift/pull/1051/files?w=1).